### PR TITLE
Edit neighborhood route authentication and tests; resolve #70

### LINKED ISSUE #70: Implement authorization for PUT neighborhoods/:id route
User should be logged in and admin.
Use the `userExtractor` middleware and the `isAdmin` function within the `routeHelpers.ts` module.
You can use the Neighborhoods delete route as an example.

### PR DESCRIPTION
**Pull request by nina-dk**

### CODE REVIEW

File: src/utils/routeHelpers.ts [hunk 1]
```diff
@@ -161,10 +161,15 @@ const isAdmin = async (loggedUserID: number, neighborhoodID: number): Promise<bo
   return (neighborhood.admin_id === loggedUserID);
 };
 
+const isLoggedInAdmin = async (req: CustomRequest) => (
+  (req.user !== undefined) && await isAdmin(req.user!.id, Number(req.params.id))
+);
```
Comment (radu-constantin) [inline src/utils/routeHelpers.ts:166]: Good thinking with function! Makes the code a lot cleaner.

**Review by radu-constantin — APPROVED** [review]
Great job as usual Antonina!

File: src/utils/routeHelpers.ts (hunk 1 above)
Comment (RobertRodes) [reply src/utils/routeHelpers.ts:166]: Nice bit of code, Antonina. Well conceived and well executed.

**Review by RobertRodes — APPROVED** [review]

File: src/utils/routeHelpers.ts (hunk 1 above)
Comment (nina-dk) [reply src/utils/routeHelpers.ts:170]: thanks guys!!

**Pull request merged**